jax/_src/scipy/cluster/vq.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import operator
+
+import scipy.cluster.vq
+import textwrap
+
+from jax import vmap
+from jax._src.numpy.util import _wraps, _promote_args_inexact
+from jax._src.numpy.lax_numpy import argmin
+from jax._src.numpy.linalg import norm
+
+
+_no_chkfinite_doc = textwrap.dedent("""
+Does not currently support the Scipy argument ``check_finite=True``,
+because `jax.numpy.asarray_chkfinite` does not exist at the moment.
+""")
+
+
+@_wraps(scipy.cluster.vq.vq, lax_description=_no_chkfinite_doc, skip_params=('check_finite',))
+def vq(obs, code_book, check_finite=True):
+    # note: _promote_args_inexact perform _check_arraylike and broadcasting
+    obs, code_book = _promote_args_inexact("vq", obs, code_book)
+    assert len(code_book.shape) == 2

```

Comment:
`assert code_book.ndim == 2`

Also, scipy's version appears to also accept 1-dimensional inputs

Reply:
I will dig into `scipy` behavior, and use `ndim` instead of `len` of `shape`, I just forget it, thanks!